base/sparse/sparsematrix.jl

```diff
@@ -1390,7 +1390,77 @@ function one{T}(S::SparseMatrixCSC{T})
     speye(T, m)
 end
 
-## Unary arithmetic and boolean operators
+
+## Broadcast operations involving a single sparse matrix and possibly broadcast scalars
+
+function broadcast{Tf}(f::Tf, A::SparseMatrixCSC)
+    fofzero = f(zero(eltype(A)))
+    fpreszero = fofzero == zero(fofzero)
+    return fpreszero ? _broadcast_zeropres(f, A) : _broadcast_notzeropres(f, A)
```

Comment:
Shouldn't you pass `fofzero` to `_broadcast_notzeropres` so that it doesn't need to be recomputed?